app/interactions/matches/create_match.rb

```diff
@@ -0,0 +1,115 @@
+class CreateMatch < ActiveInteraction::Base
+  integer :number_of_players, default: 6
+
+  validates_numericality_of :number_of_players,
+                            greater_than_or_equal_to: 3,
+                            less_than_or_equal_to: 6
+
+  def execute
+    ActiveRecord::Base.transaction do
+      match = Match.create!
+      match.create_board!
+      create_players(match, number_of_players)
+      # create_power_pool(match)
+      # create_decks(match, use_tides_of_battle_cards)
+      match
+    end
+  end
+
+  private
+
+  # 3 players: Baratheon Lannister Stark
+  # 4 players: Baratheon Lannister Stark Greyjoy
+  # 5 players: Baratheon Lannister Stark Greyjoy Tyrell
+  # 6 players: Baratheon Lannister Stark Greyjoy Tyrell Martell
+  # (Rulebook, page 28)
+  def create_players(match, number_of_players)
+    house_names = Player::HOUSES.first(number_of_players)
+    house_names.each do |house_name|
+      player = match.players.create!(match: match, house: house_name)
+      distribute_player_things(player, match.board)
+    end
+  end
+
+  def distribute_player_things(player, board)
+    create_orders(player, board)
+    create_power_tokens(player)
+    compose(CreateStartingUnits, player: player, board: board)
+    send("create_#{player.house.downcase}_cards", player)
+  end
+
+  def create_tracks(match, board, players)
+    [
+      WildlingsTrack,
+      VictoryTrack,
+      # SupplyTrack,               # TODO
+      # RoundTrack,                # TODO
+      # KingsCourtInfluenceTrack,  # TODO
+      # IronThroneInfluenceTrack,  # TODO
+      # FiefdomsInfluenceTrack     # TODO
+    ].each do |track|
+      track.setup!(match, board, players)
+    end
+  end
+
+  def create_decks(match, use_tides_of_battle_cards)
+    # TODO
+    # create_westeros_deck_one(match)
+    # create_westeros_deck_two(match)
+    # create_westeros_deck_three(match)
+    # create_wildlings_deck(match)
+    # create_tides_of_battle_deck(match) if use_tides_of_battle_cards
+  end
+
+  def create_power_pool(match)
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Unused method argument - `match`. If it's necessary, use `_` or `_match` as an argument name to indicate that it won't be used. You can also write as `create_power_pool(*)` if you want the method to accept any arguments but don't care about them.](https://www.codacy.com/app/Got-Board-Team/got-board-api/file/2383240825/issues/source?bid=3216569&fileBranchId=3216571#l64)